tck/src/main/java/org/eclipse/microprofile/concurrency/tck/MPConfigTest.java

```diff
@@ -405,6 +411,81 @@ public void overrideContextPropagationForThreadContextParameterWithConfig() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+    /**
+     * Verify MicroProfile Config overrides the propagated attribute to enabled
+     * an implied clear of all remaining.
+     * @throws TimeoutException 
+     * @throws ExecutionException 
+     * @throws InterruptedException 
+     */
+    @Test
+    public void overrideContextPropagationForThreadContextWithImpliedCleared() throws InterruptedException, ExecutionException, TimeoutException {
+        ExecutorService unmanagedThreads = null;
+        
+        // Expected config is propagated=Buffer; cleared=Remaining; unchanged=Label
+        Assert.assertNotNull(clearAllRemainingThreadContext,
+                "Unable to inject ThreadContext qualified by NamedInstance. Cannot run test.");
+        
+        int originalPriority = Thread.currentThread().getPriority();     
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 3 ? 2 : 3;
+            Thread.currentThread().setPriority(newPriority);
+            Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-A"));
+            Label.set("clearUnspecifiedContexts-test-label-A");
+
+            Callable<Integer> callable = clearAllRemainingThreadContext.contextualCallable(() -> {
+                try {
+                    Assert.assertEquals(Buffer.get().toString(), "clearUnspecifiedContexts-test-buffer-A",
+                            "Context type was not propagated to contextual action.");
+
+                    Assert.assertEquals(Label.get(), "",
+                            "Context type was not left unchanged by contextual action.");
+
+                    Label.set("clearUnspecifiedContexts-test-label-B");
+
+                    return Thread.currentThread().getPriority();
+                }
+                finally {
+                    // Restore original values
+                    Buffer.set(null);
+                    Label.set(null);
+                    Thread.currentThread().setPriority(originalPriority);
+                }
+            });
+
+            unmanagedThreads = Executors.newFixedThreadPool(5);
```

Comment:
As stated earlier, I think the extra thread pool gets in the way of the test logic rather than helping it.